### ISSUE TITLE
Capability fails on unknown capabilities

### ISSUE THREAD

**Issue by samjsong** [docker-java/docker-java#1740]
In certain environments, we're seeing this error:
```
Error [com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `com.github.dockerjava.api.model.Capability` from String "net_admin": not one of the values accepted for Enum class: [ALL, SYS_BOOT, DAC_OVERRIDE, NET_RAW, BLOCK_SUSPEND, FOWNER, IPC_LOCK, IPC_OWNER, SYS_PACCT, NET_BIND_SERVICE, WAKE_ALARM, FSETID, DAC_READ_SEARCH, SYS_CHROOT, SYS_RAWIO, SYS_ADMIN, KILL, MAC_ADMIN, SYS_RESOURCE, CHOWN, SETPCAP, SYS_PTRACE, NET_ADMIN, SETFCAP, SYS_NICE, LINUX_IMMUTABLE, AUDIT_CONTROL, LEASE, AUDIT_WRITE, SYS_MODULE, MKNOD, SYSLOG, MAC_OVERRIDE, SYS_TIME, SETGID, SETUID, SYS_TTY_CONFIG, NET_BROADCAST]
```

Our docker container inspect output shows:
```
"CapAdd": [
  "net_admin",
  "net_raw"
],
```

**Comment by bsideup** [comment]
Are you using Podman?

**Comment by samjsong** [comment]
We've opened this issue as a proxy for a customer, but my understanding is that they are using Nomad.

In the past we've seen https://github.com/docker-java/docker-java/issues/1488 not on Podman but on Docker Swarm. But that issue is currently not a priority and we're looking to fix the case sensitivity instead.

**Comment by samjsong** [comment]
Hi team, any updates here? Is the opened PR undergoing review?

**Comment by samjsong** [comment]
Bump. We'd appreciate getting review on https://github.com/docker-java/docker-java/pull/1741 please.

**Comment by lei-sumo** [comment]
Hi @bsideup Could anyone review this? We have customers waiting on this fix.

**Comment by bsideup** [comment]
Hi @lei-sumo @samjsong,

The PR is failing on CI. Please fix it if you want it to be accepted.

I also wanted to point out that you can always tune `ObjectMapper` by implementing `DockerClientConfig#getObjectMapper`.

Here is an example:
```java
DockerClientConfig original = ...;
new DockerClientConfigDelegate(original) {

    ObjectMapper objectMapper = getObjectMapper().copy()
        .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES);

    @Override
    public ObjectMapper getObjectMapper() {
        return objectMapper;
    }
};
```

**Comment by samjsong** [comment]
Thank you for the response! We weren't aware of `DockerClientConfigDelegate`, and we appreciate you providing us with an example. This looks like it will probably work for our use-case. We will try it and get back to you.

**Comment by cvandal** [comment]
Hi @samjsong and @bsideup,

We're currently in the process of implementing Nomad and have bumped into this same problem. @samjsong I noticed you haven't committed anything to your PR since December; did you manage to solve the problem another way?

I took a look at the CI failures from @samjsong's PR and I'm wondering if they're due to the change or something else because I'm seeing errors like `Expected: (an instance of java.io.IOException and exception with message a string containing "The Auth Config file is empty")`
